crates/biome_js_analyze/src/lint/nursery/no_enum.rs

```diff
@@ -0,0 +1,51 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::TsEnumDeclaration;
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Disallow TypeScript enum.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// enum Foo {
+    ///     BAR = 'bar'
+    /// }
+    /// ```
+    ///
+    pub NoEnum {
+        version: "next",
+        name: "noEnum",
+        language: "ts",
+        recommended: false,
+    }
+}
+
+impl Rule for NoEnum {
+    type Query = Ast<TsEnumDeclaration>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(_ctx: &RuleContext<Self>) -> Self::Signals {
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                ctx.query().range(),
```

Comment:
It depends on your design, but you could consider passing the span directly instead of querying it here

Reply:
do you mean simply

```rs
let span = ctx.query().range();

Some(
  RuleDiagnostic::new(
    rule_category!(),
    span,
    // ...
```
or move it to the `run` method and and store the span inside the `State` of the rule (like some other rules do)? would there be any advantages I don't know of in doing so?

Reply:
I don't think there's a need to save the span inside the state. I believe the current implementation is correct